src/extensions/averagewaveforms/AverageWaveformsView/AverageWaveformsView.tsx

```diff
@@ -1,34 +1,92 @@
-import React, { FunctionComponent, useMemo } from 'react';
+import React, { FunctionComponent, useCallback, useEffect, useMemo, useState } from 'react';
+import { FaArrowDown, FaArrowUp } from 'react-icons/fa';
+import sizeMe, { SizeMeProps } from 'react-sizeme';
 import SortingUnitPlotGrid from '../../common/SortingUnitPlotGrid';
+import Splitter from '../../common/Splitter';
+import { ActionItem, DividerItem } from '../../common/Toolbars';
 import { SortingViewProps } from "../../extensionInterface";
+import AverageWaveformsToolbar from './AverageWaveformsToolbar';
 import AverageWaveformView2 from './AverageWaveformView2';
 
-const AverageWaveformsView: FunctionComponent<SortingViewProps> = (props) => {
+export type AverageWaveformAction = ActionItem | DividerItem
+
+
+const TOOLBAR_INITIAL_WIDTH = 55
+
+const AverageWaveformsView: FunctionComponent<SortingViewProps & SizeMeProps> = (props) => {
     const {recording, sorting, selection, selectionDispatch} = props
     const boxHeight = 250
     const boxWidth = 180
-
     const noiseLevel = (recording.recordingInfo || {}).noise_level || 1  // fix this
+    const [scalingActions, setScalingActions] = useState<AverageWaveformAction[] | null>(null)
     const unitComponent = useMemo(() => (unitId: number) => (
         <AverageWaveformView2
             {...{sorting, recording, unitId, selection, selectionDispatch}}
             width={boxWidth}
             height={boxHeight}
             noiseLevel={noiseLevel}
+            customActions={scalingActions || []}
         />
-    ), [sorting, recording, selection, selectionDispatch, noiseLevel])
+    ), [sorting, recording, selection, selectionDispatch, noiseLevel, scalingActions])
 
+    const width = props.size.width;
+    const height = 650 // hard-coded as per TimeseriesForRecordingView.tsx
 
-    // useCheckForChanges('AverageWaveformsView', props)
+    const _handleScaleAmplitudeUp = useCallback(() => {
+        selectionDispatch({type: 'ScaleAmpScaleFactor', multiplier: 1.15})
+    }, [selectionDispatch])
+    const _handleScaleAmplitudeDown = useCallback(() => {
+        selectionDispatch({type: 'ScaleAmpScaleFactor', multiplier: 1 / 1.15})
+    }, [selectionDispatch])
```

Comment:
Defines the scaling operation so it can be passed into individual units that have layers (to listen to the keystrokes) and into the toolbar (for button clicks).